core/runtime/jsruntime.rs

```diff
@@ -1965,7 +1976,7 @@ impl JsRuntime {
           break;
         };
         // TODO(mmastrac): If this task is really errored, things could be pretty bad
-        let (promise_id, op_id, resp) = item.unwrap();
+        let PendingOp(promise_id, op_id, resp, metrics_event) = item.unwrap();
```

Comment:
Is it really free if each now now returns `metrics_event`?

Reply:
It's effectively free -- we're just setting an extra boolean in a tuple struct here. Once we have a chance to revisit this async dispatch, it will likely turn out to be cheaper.